hooks/headless/useChatMessages.tsx

```diff
@@ -1,54 +1,201 @@
-import { createReference } from "@medplum/core";
-import { Communication } from "@medplum/fhirtypes";
-import { useMedplum } from "@medplum/react-hooks";
-import { useEffect, useState } from "react";
+import { createReference, getReferenceString, ProfileResource } from "@medplum/core";
+import { Bundle, Communication, Reference } from "@medplum/fhirtypes";
+import { useMedplum, useSubscription } from "@medplum/react-hooks";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
 
-import type { ChatMessage } from "@/types/chat";
-import { formatTimestamp } from "@/utils/datetime";
+import { ChatMessage } from "@/types/chat";
 
-export function useChatMessages(threadId: string) {
+function upsertCommunications(
```

Comment:
This function name led me to believe it performed an async operation because I was reminded of the `medplum.upsertResource`. Not sure if it could mislead others.